assistedInstaller.py

```diff
@@ -184,6 +185,16 @@ def get_ai_host(self, name: str) -> Optional[AssistedClientHostInfo]:
                 return h
         return None
 
+    def get_ai_host_id_by_ip(self, infra_env: str, ip: str) -> Optional[AssistedClientHostInfo]:
+        id = self.get_infra_env_id(infra_env)
+        for h in filter(lambda x: "inventory" in x and x["infra_env_id"] == id, self.list_hosts()):
+            nics = json.loads(h["inventory"]).get("interfaces")
+            addresses: list[str] = sum((nic["ipv4_addresses"] for nic in nics), [])
+            stripped_addresses = [a.split("/")[0] for a in addresses]
+            if ip in stripped_addresses:
+                return AssistedClientHostInfo(h["requested_hostname"], h["id"], h["status"], h["inventory"])
```

Comment:
Why recreate the host hest? you already have `h`. Isn't that the same thing? Also, why do we need to pass in infra_env_id?

```
def get_ai_host_by_ip(self, ip: str) ->  Optional[AssistedClientHostInfo]:
    for h in self.list_ai_hosts():
        nics = json.loads(h.inventory).get("interfaces")
        addresses = sum((nic["ipv4_addresses"] for nic in nics), [])
        stripped_addresses = [a.split("/")[0] for a in addresses]
        if ip in stripped_addresses:
            return h
    return None
```

Reply:
The `list_ai_hosts` helper was added after the original code, but I have used your suggestion.